nats/src/main/java/io/micronaut/nats/intercept/NatsIntroductionAdvice.java

```diff
@@ -107,6 +115,23 @@ public Object intercept(MethodInvocationContext<Object, Object> context) {
                 Argument<?> bodyArgument = findBodyArgument(method).orElseThrow(
                         () -> new NatsClientException("No valid message body argument found for method: " + method));
 
+                Headers methodHeaders = new Headers();
+
+                List<AnnotationValue<MessageHeader>> headerAnnotations =
+                        method.getAnnotationValuesByType(MessageHeader.class);
+                Collections.reverse(headerAnnotations); //set the values in the class first so methods can override
+                headerAnnotations.forEach(header -> {
+                    String name = header.get("name", String.class).orElse(null);
+                    String value = header.getValue(String.class).orElse(null);
+
+                    if (StringUtils.isNotEmpty(name) && StringUtils.isNotEmpty(value)) {
+                        if (!methodHeaders.containsKey(name)) {
+                            methodHeaders.put(name, new ArrayList<>());
+                        }
+                        methodHeaders.put(name, value);
```

Comment:
Is this code corrrect? it seems the array list is always overridden?

Reply:
Removed the if condition as it was useless.